CSharp/BatchExplorer/Service/BatchService.cs

```diff
@@ -201,13 +201,48 @@ public Task<CloudPool> GetPoolAsync(string poolId)
             int targetDedicated,
             TimeSpan? timeout,
             ComputeNodeDeallocationOption? deallocationOption)
-        {
+        {
+            var pool = await this.GetPoolAsync(poolId);
+            if (pool.AutoScaleEnabled.HasValue && pool.AutoScaleEnabled.Value)
+            {
+                await this.Client.PoolOperations.DisableAutoScaleAsync(poolId);
+                if (!await this.WaitForResizingOperationToFinish(poolId))
+                {
+                    throw new InvalidOperationException("Autoscale disable operation is already in progress. Please try again after sometime to Resize the pool.");

```

Comment:
The operation will always hit an error if we do not wait for a few seconds. This could be annoying or confusing for the user everytime he performs this action. Please note that it is just a "Disable auto scale" operation which should complete within expected time. The actual resize operation happens after the wait operation has finished.

I have to suggestions on this point.

1) As paselem said, we can make this a background operation so that it doesn't block the UI thread. User can track the status from the operation's window. This still requires a wait operation in the background otherwise, it would always fail. (Please see the code, I have already made this change)

2) Second option is to disable the "Fixed radio button" until user explicitly clicks on "Disable Auto scaling" (New button)

Please advise which approach is better? Suggestion 1 has already incorporated as it was a small change.